src/lib/mina_ledger/ledger.ml

```diff
@@ -201,6 +277,47 @@ module Ledger_inner = struct
     let _base, mask = create_ephemeral_with_base ~depth () in
     mask
 
+  type converting_config =
+    { primary_directory_name : string option
+    ; converting_directory_name : string option
+    }
+
+  let default_converting_directory_name primary_directory_name =
+    primary_directory_name ^ "_converting"
+
+  let converting_directory_name ~cfg ~primary_directory =
+    Option.first_some cfg.converting_directory_name
+    @@ Option.map primary_directory ~f:default_converting_directory_name
+
+  let empty_converting_config : converting_config =
+    { primary_directory_name = None; converting_directory_name = None }
+
+  let create_converting ?(cfg = empty_converting_config) ~depth () =
+    let open Or_error.Let_syntax in
+    let db1 = Db.create ?directory_name:cfg.primary_directory_name ~depth () in
+    let db2 =
+      let directory_name =
+        converting_directory_name ~cfg ~primary_directory:(Db.get_directory db1)
+      in
+      Unstable_db.create ?directory_name ~depth ()
+    in
+    let%bind converting_ledger =
+      if Unstable_db.num_accounts db2 = 0 then
+        Or_error.return @@ Converting_ledger.create_with_migration db1 db2
+      else if not (Db.num_accounts db1 = Unstable_db.num_accounts db2) then (
+        (* TODO: ensure 2 dbs have exact same accounts *)
```

Comment:
Noted

Reply:
Well, we have test coverage for this so it's not that fatal, though

Reply:
To respond to https://github.com/MinaProtocol/mina/pull/17379#discussion_r2144147336 here - yes, this is insufficient as a desync check. A converting merkle tree is two databases and we write to them separately, so it should be possible for the daemon to crash between database updates, which would cause a desync. I think you're right, that that is only detectable with a full equality check (up to account conversion) between the databases.

Reply:
I think part of the slowness of migration is all the hashing that needs to be done to populate the database, not necessarily the actual account conversion. If the `Unstable.of_stable` conversion function is expected to be fast, then we might be able to do this integrity check right here. @georgeee do you know how heavy the real conversion function is likely to be?

Reply:
> I think part of the slowness of migration is all the hashing that needs to be done to populate the database, not necessarily the actual account conversion.

This is my expectation as well

> do you know how heavy the real conversion function is likely to be?

We expect conversion to be trivial, i.e. not using any cryptographic operations at all.